src/System.Threading.AccessControl/src/System/Threading/EventWaitHandleAcl.cs

```diff
@@ -0,0 +1,110 @@
+        /// <exception cref="ArgumentOutOfRangeException">The <paramref name="mode" /> enum value was out of legal range.</exception>
+        /// <exception cref="WaitHandleCannotBeOpenedException">An <see cref="EventWaitHandle" /> with system-wide name <paramref name="name" /> cannot be created. An <see cref="EventWaitHandle" /> of a different type might have the same name.</exception>
+        public static EventWaitHandle Create(bool initialState, EventResetMode mode, string name, out bool createdNew, EventWaitHandleSecurity eventSecurity)
+        {
+            if (name != null && name.Length > Interop.Kernel32.MAX_PATH)
+            {
+                throw new ArgumentException(SR.Format(SR.Argument_WaitHandleNameTooLong, name));
+            }
+
+            if (eventSecurity == null)
+            {
+                throw new ArgumentNullException(nameof(eventSecurity));
+            }
+
+            bool isManualReset;
+            switch (mode)
+            {
+                case EventResetMode.ManualReset:
+                    isManualReset = true;
+                    break;
+                case EventResetMode.AutoReset:
+                    isManualReset = false;
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(mode), SR.Format(SR.ArgumentOutOfRange_Enum));
+            };
+
+            SafeWaitHandle handle = CreateWaitHandle(initialState, isManualReset, name, out createdNew, eventSecurity);
+
+            try
+            {
+                // This constructor is private
+                // return new EventWaitHandle(handle);
+                return null;
+            }
+            catch
+            {
+                handle.Dispose();
+                throw;
+            }
+        }
+
+
+        private static unsafe SafeWaitHandle CreateWaitHandle(bool initialState, bool isManualReset, string name, out bool createdNew, EventWaitHandleSecurity security)
+        {
+            SafeWaitHandle handle;
+
+            fixed (byte* pSecurityDescriptor = security.GetSecurityDescriptorBinaryForm())
+            {
+                var secAttrs = new Interop.Kernel32.SECURITY_ATTRIBUTES
+                {
+                    nLength = (uint)sizeof(Interop.Kernel32.SECURITY_ATTRIBUTES),
+                    lpSecurityDescriptor = (IntPtr)pSecurityDescriptor
+                };
+
+                handle = Interop.Kernel32.CreateEvent(ref secAttrs, isManualReset, initialState, name);
+                ValidateHandle(handle, name, out createdNew);
+            }
+
+            return handle;
+        }
+
+        private static void ValidateHandle(SafeWaitHandle handle, string name, out bool createdNew)
+        {
+            createdNew = false;
+
+            if (handle.IsInvalid)
+            {
+                // We call this in netfx. Is it still needed at this point?
```

Comment:
Will remove this comment and/or the next line depending on the answer to this question.

Reply:
Ok thanks. I'll keep the code and remove the comment.